File: app/src/main/java/com/example/mutidemo/mvvm/WeatherAdapter.java
package com.example.mutidemo.mvvm;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.mutidemo.R;
import com.example.mutidemo.mvvm.model.WeatherModel;

import java.util.List;

public class WeatherAdapter extends RecyclerView.Adapter<WeatherAdapter.WeatherViewHolder> {

    private final LayoutInflater layoutInflater;
    private final List<WeatherModel.ResultBean.WeatherBean.DailyBean> dataRows;

    public WeatherAdapter(Context context, List<WeatherModel.ResultBean.WeatherBean.DailyBean> dataRows) {
        this.dataRows = dataRows;
        this.layoutInflater = LayoutInflater.from(context);
    }

    @Override
    public int getItemCount() {
        return dataRows == null ? 0 : dataRows.size();
    }

    @NonNull
    @Override
    public WeatherAdapter.WeatherViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new WeatherViewHolder(
                layoutInflater.inflate(R.layout.item_weather_rv, parent, false)
        );
    }

    @Override
    public void onBindViewHolder(@NonNull WeatherAdapter.WeatherViewHolder holder, int position) {
        WeatherModel.ResultBean.WeatherBean.DailyBean dailyBean = dataRows.get(position);
        WeatherModel.ResultBean.WeatherBean.DailyBean.DayBean dayBean = dailyBean.getDay();
        WeatherModel.ResultBean.WeatherBean.DailyBean.NightBean nightBean = dailyBean.getNight();
        switch (position) {
            case 0:
                holder.weatherWeek.setText("今天");
                break;
            case 1:
                holder.weatherWeek.setText("明天");
                break;
            default:
                holder.weatherWeek.setText(dailyBean.getWeek());
                break;
        }
        holder.weatherLow.setText(nightBean.getTemplow() + "°");
        holder.weatherHigh.setText(dayBean.getTemphigh() + "°");
    }

    static class WeatherViewHolder extends RecyclerView.ViewHolder {

        private final TextView weatherWeek;
        private final TextView weatherLow;
        private final TextView weatherHigh;

        WeatherViewHolder(@NonNull View itemView) {
            super(itemView);
            weatherWeek = itemView.findViewById(R.id.weatherWeek);
            weatherLow = itemView.findViewById(R.id.weatherLow);
            weatherHigh = itemView.findViewById(R.id.weatherHigh);
        }
    }
}

File: app/src/main/java/com/example/mutidemo/mvvm/WeatherAdapter.kt
package com.example.mutidemo.mvvm

import android.annotation.SuppressLint
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.mutidemo.R
import com.example.mutidemo.mvvm.model.DailyModel

class WeatherAdapter(
    ctx: Context?,
    private val mItemList: List<DailyModel>?
) : RecyclerView.Adapter<WeatherAdapter.WeatherViewHolder>() {

    private val inflater: LayoutInflater = LayoutInflater.from(ctx)
    override fun getItemCount(): Int {
        //不显示当天
        return mItemList?.size ?: 0
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): WeatherViewHolder {
        return WeatherViewHolder(
            inflater.inflate(
                R.layout.item_weather_rv,
                parent,
                false
            )
        )
    }

    @SuppressLint("SetTextI18n")
    override fun onBindViewHolder(holder: WeatherViewHolder, position: Int) {
        when (position) {
            0 -> {
                val dailyBean = mItemList!![position]
                val dayBean = dailyBean.day
                val nightBean = dailyBean.night
                holder.weatherWeek.text = "今天"
                holder.weatherLow.text = nightBean.templow + "°"
                holder.weatherHigh.text = dayBean.temphigh + "°"
            }
            1 -> {
                val dailyBean = mItemList!![position]
                val dayBean = dailyBean.day
                val nightBean = dailyBean.night
                holder.weatherWeek.text = "明天"
                holder.weatherLow.text = nightBean.templow + "°"
                holder.weatherHigh.text = dayBean.temphigh + "°"
            }
            else -> {
                val dailyBean = mItemList!![position]
                val dayBean = dailyBean.day
                val nightBean = dailyBean.night
                holder.weatherWeek.text = dailyBean.week
                holder.weatherLow.text = nightBean.templow + "°"
                holder.weatherHigh.text = dayBean.temphigh + "°"
            }
        }
    }

    class WeatherViewHolder constructor(itemView: View) :
        RecyclerView.ViewHolder(itemView) {
        val weatherWeek: TextView = itemView.findViewById(R.id.weatherWeek)
        val weatherLow: TextView = itemView.findViewById(R.id.weatherLow)
        val weatherHigh: TextView = itemView.findViewById(R.id.weatherHigh)
    }
}